File: python/src/Behavioral/Memento/DvdDetails.py
#//Memento (aka Token) Overview
#//One object stores another objects state.
#//DvdDetails - the Originator
#//(the class to be backed up)
#//Contains the inner class DvdMemento - the Memento

from DvdMemento import DvdMemento

class DvdDetails :

	def __init__(self,titleName,stars = [],region = 1):
		self.titleName = titleName
		self.stars = stars
		self.region = region


	def addStar(self,star) :
		self.stars.append(star)

	def starsAsString(self) :
		return ', '.join(stars)

	def formatDvdDetails(self) : 
		return "DVD: #{0} starring: {1} encoding region: {2}".format(self.titleName,self.starsAsString(), self.region);

	#   //sets current state to what DvdMemento has
	def setDvdMemento (self,memento ) :
		for k in self:
			self[k] = None
		c = memento.getState()
		for k in c:
			self[k] = c[k]

	#//save current state of DvdDetails in a DvdMemento
	def createDvdMemento (self) :
		mem = DvdMemento()
		mem.setState(self)
		mem

File: ruby/src/Behavioral/Memento/DvdDetails.rb
#//Memento (aka Token) Overview
#//One object stores another objects state.
#//DvdDetails - the Originator
#//(the class to be backed up)
#//Contains the inner class DvdMemento - the Memento

require "Behavioral/Memento/DvdMemento"

class DvdDetails

	attr_accessor :titleName,:stars,:region

	def initialize(titleName,stars = [],region = 1)
		@titleName = titleName
		@stars = stars
		@region = region
	end

	def addStar(star)
 		@stars.push(star)
	end

	def starsAsString()
 		return @stars.join(', ')
	end

	def formatDvdDetails()
 		"DVD: #{ @titleName }, starring: #{ starsAsString() }, encoding region: #{ @region }"
	end

	#   //sets current state to what DvdMemento has
	def setDvdMemento (memento )
		c = memento.getState()
		@titleName = c.titleName
		@stars = c.stars
		@region = c.region
	end

	#//save current state of DvdDetails in a DvdMemento
	def createDvdMemento ()
 		mem = DvdMemento.new(self)
		return mem
	end

end
